README
                           Rosetta-Engine-Native
---------------------------------------------------------------------------

by Darren Duncan <perl@DarrenDuncan.net>

PREFACE

This distribution features the Perl 5 module named
"Rosetta::Engine::Native"; see the file lib/Rosetta/Engine/Native.pm for
the main documentation, which is in POD format, plus the Dependencies and
Copyright.  The distribution also includes a Changes file to say what
changes already happened, and a TODO file to say what changes have yet to
happen.  Read below for how to get support or keep up to date.

KEEPING UP TO DATE

My module set is constantly under development.  The canonical copies are
all linked to through my website, "http://www.DarrenDuncan.net/", on the
page called "Perl Libraries I Made" (name subject to change).

For those of you that are interested in the bleeding edge of development, I
have all parts of this "Rosetta-Engine-Native" distribution in one or more
public Subversion repositories.  The code base is available in 2 parallel
versions at "http://svn.openfoundry.org/pugs/ext/Rosetta-Engine-Native/"
(Perl 6) and at "http://svn.utsl.gen.nz/trunk/Rosetta-Engine-Native/" (Perl
5).

You can download slightly older versions of all of my modules from the
public CPAN archives at "http://www.cpan.org/authors/id/D/DU/DUNCAND/", or
on my own web server at "http://www.DarrenDuncan.net/d/perl".

CPAN also extracts the POD from modules and maintains that documentation in
an organized and searchable fashion.  Their main search site is at
"http://search.cpan.org", and my own DUNCAND author page is at
"http://search.cpan.org/search?author=DUNCAND".

You can download my oldest tarred and gzipped distributions from BACKPAN,
at "http://history.perl.org/backpan/authors/id/D/DU/DUNCAND/", or from my
web server, at "http://www.DarrenDuncan.net/d/perl/archives/", right back
to the libdwg 1.0, first released on 2000 July 23.

SUPPORT

Currently I don't have any support arranged with other people, lists,
newsgroups, or otherwise.  Feel free to ask me if you can't figure things
out on your own, or another person whom you know has used this.  I may
start a mailing list for support issues later, so that users of my modules
can help each other with them.

FIN

Share and Enjoy!


Changes
                           Rosetta-Engine-Native
---------------------------------------------------------------------------

2006-02-01   Darren Duncan <perl@DarrenDuncan.net>

    * Rosetta-Engine-Native Release 0.1.0/0.001000 (SVN r1203).

    * This release is concurrently available in both Perl 5 and Perl 6
    versions.  The Perl 5 version is in the same distribution as this
    Changes file.  The Perl 6 version is bundled with Pugs-6.2.11 (released
    on 2006-02-01), in its ext/Rosetta-Engine-Native/ subdirectory.

    * Initial code file versions are: Native.pm 0.1.0 and en.pm 0.1.0.

    * Initial external dependencies are: 'version', 'only', 'Readonly'
    1.03, Class::Std 0.0.4, Class::Std::Utils 0.0.2, Locale::KeyedText
    1.72.0, 'Rosetta' 0.720.0, Test::More, Rosetta::Validator 0.720.0.

    * This is the first release of the Rosetta-Engine-Native distribution.
    This is the first release of any distribution to contain
    Rosetta::Engine::Native.  This is the first release of any distribution
    for which some of the existing work that became the
    Rosetta-Engine-Native distribution was in the form of executable code,
    and not just design documentation.

    * This is the initial file manifest:

        Changes
        GPL
        INSTALL
        lib/Rosetta/Engine/Native.pm
        lib/Rosetta/Engine/Native/L/en.pm
        Makefile.PL
        MANIFEST
        META.yml
        README
        t/Rosetta_Engine_Native.t
        TODO

2005-09-08   Darren Duncan <perl@DarrenDuncan.net>

    * These did not exist and were not cited by the public announcement for
    Rosetta/SQL-Routine developer release #3.

2005-04-03   Darren Duncan <perl@DarrenDuncan.net>

    * These did not exist and were not cited by the public announcement for
    SQL-Routine/Rosetta developer release #2.

2004-08-06   Darren Duncan <perl@DarrenDuncan.net>

    * These did not exist and were not cited by the public announcement for
    Rosetta/SQL-SyntaxModel developer release #1.

2002-06-07

    * This is when I started writing self-contained code components that
    were explicitly designed to enable external code that used them to work
    seamlessly on multiple database products, and hence 2002 is the start
    of my declared copyright date range for the Rosetta framework.


lib/Rosetta/Engine/Native.pm
#!perl
use 5.008001;
use utf8;
use strict;
use warnings;

# External packages used by packages in this file, that don't export symbols:
use only 'Locale::KeyedText' => '1.72.0-';
use only 'Rosetta' => '0.720.0-';

###########################################################################
###########################################################################

# Constant values used by packages in this file:
use only 'Readonly' => '1.03-';
# (None Yet)

###########################################################################
###########################################################################

{ package Rosetta::Engine::Native; # class
    use version; our $VERSION = qv('0.1.0');

    # External packages used by the Rosetta::Engine::Native class, that do export symbols:
    use only 'Class::Std' => '0.0.4-';
    use only 'Class::Std::Utils' => '0.0.2-';

    # Attributes of every Rosetta::Engine::Native object:
    # (None Yet)

###########################################################################



###########################################################################

} # class Rosetta::Engine::Native

###########################################################################
###########################################################################

1; # Magic true value required at end of a reuseable file's code.
__END__

=pod

=encoding utf8

=head1 NAME

Rosetta::Engine::Native -
Self-contained reference implementation of a Rosetta Engine

=head1 VERSION

This document describes Rosetta::Engine::Native version 0.1.0.

=head1 SYNOPSIS

I<This documentation is pending.>

=head1 DESCRIPTION

I<This documentation is pending.>

=head1 INTERFACE

I<This documentation is pending; this section may also be split into several.>

=head1 DIAGNOSTICS

I<This documentation is pending.>

=head1 CONFIGURATION AND ENVIRONMENT

I<This documentation is pending.>

=head1 DEPENDENCIES

This file requires any version of Perl 5.x.y that is at least 5.8.1.

It also requires the Perl 5 packages L<version> and L<only>, which would
conceptually be built-in to Perl, but aren't, so they are on CPAN instead.

It also requires these Perl 5 packages that are on CPAN:
L<Readonly-(1.03...)|Readonly>.

It also requires these Perl 5 packages that are on CPAN:
L<Class::Std-(0.0.4...)|Class::Std>,
L<Class::Std::Utils-(0.0.2...)|Class::Std::Utils>.

It also requires these Perl 5 classes that are on CPAN:
L<Locale::KeyedText-(1.72.0...)|Locale::KeyedText> (for error messages),
L<Rosetta-(0.720.0...)|Rosetta>.

=head1 INCOMPATIBILITIES

None reported.

=head1 SEE ALSO

I<This documentation is pending.>

=head1 BUGS AND LIMITATIONS

I<This documentation is pending.>

=head1 AUTHOR

Darren R. Duncan (C<perl@DarrenDuncan.net>)

=head1 LICENCE AND COPYRIGHT

This file is part of the Rosetta database portability library, though it is
distributed separately from L<Rosetta> itself.

Rosetta is Copyright (c) 2002-2006, Darren R. Duncan.  All rights reserved.
Address comments, suggestions, and bug reports to C<perl@DarrenDuncan.net>,
or visit L<http://www.DarrenDuncan.net/> for more information.

Rosetta is free software; you can redistribute it and/or modify it under
the terms of the GNU General Public License (GPL) as published by the Free
Software Foundation (L<http://www.fsf.org/>); either version 2 of the
License, or (at your option) any later version.  You should have received a
copy of the GPL as part of the Rosetta distribution, in the file named
"GPL"; if not, write to the Free Software Foundation, Inc., 51 Franklin St,
Fifth Floor, Boston, MA  02110-1301, USA.

Linking Rosetta statically or dynamically with other components is making a
combined work based on Rosetta.  Thus, the terms and conditions of the GPL
cover the whole combination.  As a special exception, the copyright holders
of Rosetta give you permission to link Rosetta with other free software
components, as defined by the Free Software Foundation at
L<http://www.gnu.org/philosophy/free-sw.html>.  You may copy and distribute
such a system following the terms of the GPL for Rosetta and the licenses
of the other components concerned, provided that you include the source
code of the other components when and as the GPL requires distribution of
source code.  However, for an additional fee, the copyright holders of
Rosetta can sell you an alternate, limited license that allows you to link
Rosetta with non-free software components.

Any versions of Rosetta that you modify and distribute must carry prominent
notices stating that you changed the files and the date of any changes, in
addition to preserving this original copyright notice and other credits.
Rosetta is distributed in the hope that it will be useful, but WITHOUT ANY
WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
FOR A PARTICULAR PURPOSE.  However, for an additional fee, the copyright
holders of Rosetta can sell you a warranty for it.

While it is by no means required, the copyright holders of Rosetta would
appreciate being informed any time you create a modified version of Rosetta
that you are willing to distribute, because that is a practical way of
suggesting improvements to the standard version.

=head1 ACKNOWLEDGEMENTS

None yet.

=cut


lib/Rosetta/Engine/Native/L/en.pm
#!perl
use 5.008001;
use utf8;
use strict;
use warnings;

###########################################################################
###########################################################################

# Constant values used by packages in this file:
use only 'Readonly' => '1.03-';
Readonly my %TEXT_STRINGS => (
);

###########################################################################
###########################################################################

{ package Rosetta::Engine::Native::L::en; # module
    use version; our $VERSION = qv('0.1.0');
    sub get_text_by_key {
        my (undef, $msg_key) = @_;
        return $TEXT_STRINGS{$msg_key};
    }
} # module Rosetta::Engine::Native::L::en

###########################################################################
###########################################################################

1; # Magic true value required at end of a reuseable file's code.
__END__

=pod

=encoding utf8

=head1 NAME

Rosetta::Engine::Native::L::en -
Localization of Rosetta::Engine::Native for English

=head1 VERSION

This document describes Rosetta::Engine::Native::L::en version 0.1.0.

=head1 SYNOPSIS

I<This documentation is pending.>

=head1 DESCRIPTION

I<This documentation is pending.>

=head1 INTERFACE

I<This documentation is pending; this section may also be split into several.>

=head1 DIAGNOSTICS

I<This documentation is pending.>

=head1 CONFIGURATION AND ENVIRONMENT

I<This documentation is pending.>

=head1 DEPENDENCIES

This file requires any version of Perl 5.x.y that is at least 5.8.1.

It also requires the Perl 5 packages L<version> and L<only>, which would
conceptually be built-in to Perl, but aren't, so they are on CPAN instead.

It also requires these Perl 5 packages that are on CPAN:
L<Readonly-(1.03...)|Readonly>.

=head1 INCOMPATIBILITIES

None reported.

=head1 SEE ALSO

I<This documentation is pending.>

=head1 BUGS AND LIMITATIONS

I<This documentation is pending.>

=head1 AUTHOR

Darren R. Duncan (C<perl@DarrenDuncan.net>)

=head1 LICENCE AND COPYRIGHT

This file is part of the Rosetta database portability library, though it is
distributed separately from L<Rosetta> itself.

Rosetta is Copyright (c) 2002-2006, Darren R. Duncan.

See the LICENCE AND COPYRIGHT of L<Rosetta::Engine::Native> for details.

=head1 ACKNOWLEDGEMENTS

The ACKNOWLEDGEMENTS in L<Rosetta::Engine::Native> apply to this file too.

=cut


Makefile.PL
#!perl
use 5.008001;
use utf8;
use strict;
use warnings;

if ($] < 5.008008) {
    warn q{
---------------------------------------------------------------------------
  This distribution prefers the newest Perl 5.x.y version, which is 5.8.8
  at the time of this writing, but Perl versions down to 5.8.1 will not be
  blocked and should likely continue to work for awhile.
  You can also ignore any warnings during installation about the use of
  "=encoding utf8" in the pod documentation.
  Continuing in 3 seconds...
---------------------------------------------------------------------------
};
    sleep 3;
}

use ExtUtils::MakeMaker;

WriteMakefile(
    'NAME' => 'Rosetta::Engine::Native',
    'VERSION_FROM' => 'lib/Rosetta/Engine/Native.pm', # finds $VERSION
    'ABSTRACT' => 'Self-contained reference implementation of a Rosetta Engine',
    'AUTHOR' => 'Darren R. Duncan <perl@DarrenDuncan.net>',
#    'LICENSE' => 'gpl',
    'PREREQ_PM' => {
        # These mainly add features to Perl 5 that are built-in to Perl 6.
        'version' => 0,
        'only' => 0,
        'Readonly' => '1.03',
        'Class::Std' => '0.0.4',
        'Class::Std::Utils' => '0.0.2',
        'Locale::KeyedText' => '1.72.0',
        'Rosetta' => '0.720.0',
        # These mainly are for the test suite.
        'Test::More' => 0,
        'Rosetta::Validator' => '0.720.0',
    },
);

1; # Magic true value required at end of a reuseable file's code.
